core/src/main/java/ai/timefold/solver/core/impl/domain/variable/listener/support/VariableListenerSupport.java

```diff
@@ -57,6 +60,7 @@ public static <Solution_> VariableListenerSupport<Solution_> create(InnerScoreDi
                 .toList();
         this.listVariableDescriptor = scoreDirector.getSolutionDescriptor().getListVariableDescriptor();
         this.listVariableEventList = new ArrayList<>();
+        this.unassignedValueWithEmptyInverseEntitySet = new HashSet<>();
```

Comment:
Arguably this should be an identity hash set; no entities should be considered equal here. (The class doesn't exist in the JDK, but it can be created by converting `IdentityHashMap`.)

Reply:
Also however, for reproducibility, it should be a _linked_ hash set. (So that iteration order is stable.) This creates a problem, because there is no way to have a linked identity hash set, unless you implement one from scratch.

Reply:
Finally, the better place for this information would be `ExternalizedListVariableStateSupply`, which already keeps all information about list variables. It doesn't have a collection of unassigned elements, but if we're adding it somewhere, it should be there. `VariableListenerSupport` can still do what it does, but it should read the information from the supply.